src/Altinn.App.Core/Features/Correspondence/Builder/CorrespondenceAttachmentBuilder.cs

```diff
@@ -0,0 +1,104 @@
+using Altinn.App.Core.Features.Correspondence.Models;
+
+namespace Altinn.App.Core.Features.Correspondence.Builder;
+
+/// <summary>
+/// Builder factory for creating <see cref="CorrespondenceAttachment"/> objects
+/// </summary>
+public class CorrespondenceAttachmentBuilder : CorrespondenceBuilderBase, ICorrespondenceAttachmentBuilder
+{
+    private string? _filename;
+    private string? _name;
+    private string? _sendersReference;
+    private string? _dataType;
+    private Stream? _data;
```

Comment:
I have no preference on the interim storage for the data stream here. The builder and the record it targets both use `Stream` internally and serialize to `StreamContent`. 

Happy to modify `CorrespondenceAttachment`as required and add whatever builder overloads seem reasonable.

Reply:
In my current understanding of C# I strongly prefer `byte[]` or `ReadOnlyMemory<byte>` to `Stream`.

`Stream` is really awkward to work with, and most likely you end up needing multiple copies in memory to read muliple times. If we really have a case where it makes sense to process bytes in a streaming fashion we should use IPipeReader.

Reply:
`ReadOnlyMemory<byte>` sounds good to me considering `IInstanceDataAccessor` uses it and it seems like that's likely where app devs will be getting their input from

Reply:
Just FYI, I ended up going with `ReadOnlyMemory<byte>` as the internal storage for this data, which has a nice implicit conversion from `byte[]`. 

The serialisation now uses `ReadOnlyMemoryContent` instead of `StreamContent`.